File: flutter/hydroponics_framework/lib/datamodels/recipes/phase/manual_phase_model.dart
import 'package:hydroponics_framework/datamodels/recipes/phase/phase_model.dart';
import 'package:hydroponics_framework/datamodels/recipes/phase/phase_type.dart';

class ManualPhaseModel extends PhaseModel {

  const ManualPhaseModel({
    required super.phaseId,
    required super.name,
    required super.description,
    required super.duration
  });

  @override
  Map<String, dynamic> toJson() {
    Map<String, dynamic> json = {};
    json["type"] = PhaseType.manualPhase.name;
    json["phaseId"] = phaseId;
    json["name"] = name;
    json["description"] = description;
    json["duration"] = duration;
    return json;
  }

  static ManualPhaseModel fromJson(Map<String, dynamic> json) {
    return ManualPhaseModel(
        phaseId : json["phaseId"],
        name : json["name"],
        description: json["description"],
        duration : json["duration"]
    );
  }
}

File: kotlin/hydroponics-host/hydroponics-host/src/main/kotlin/io/greenglass/hydroponics/application/models/recipe/phase/ManualPhaseModel.kt
package io.greenglass.hydroponics.application.models.recipe.phase

import org.neo4j.graphdb.Label
import org.neo4j.graphdb.Node
import org.neo4j.graphdb.Transaction
import io.greenglass.host.application.neo4j.GraphNode

class ManualPhaseModel(
    type : PhaseType,
    phaseId : String,
    name : String,
    description : String,
    duration : Int
) : GraphNode, PhaseModel(type, phaseId,name,description, duration) {
    constructor(node : Node) : this(
        PhaseType.ManualPhase,
        node.getProperty("phaseId") as String,
        node.getProperty("name") as String,
        node.getProperty("description") as String,
        node.getProperty("duration") as Int
    )
    override fun toNode(tx : Transaction) : Node {
        val node = tx.createNode(phaseLabel, manualPhaseLabel)
        node.setProperty("phaseId", this.phaseId)
        node.setProperty("name", this.name)
        node.setProperty("description", this.description)
        node.setProperty("duration", this.duration)
        return node
    }

    companion object Graph {
        val manualPhaseLabel = Label.label("ManualPhase")
    }
}
